### PR TITLE
test: use kata 3.6.0 for testing

### PR DESCRIPTION
**Pull request by littlejawa**
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:

Update the version of kata to test with - using the latest 3.6.0 version. 
Also make kata test specific changes (affecting only the fedora-kata job).

Details:
- move to kata 3.6.0
- enable hooks tests, as the fix for it was released with 3.5.0
- make sure the storage.overlay.skip_mount_home flag is set to true (problem introduced with kata 3.3.0)
- enable debug logs for kata
- cleanup lingering kata processes on teardown
- add an integration test specific to kata, testing we're actually using the kata runtime. Useful to validate config/script changes in the CI

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: contrib/test/ci/build/kata.yml
Comment (kwilczynski) [inline contrib/test/ci/build/kata.yml:70]: A typo? :smile: 

```suggestion
    - name: Edit Kata Containers config to enable debug log level
```

Comment (littlejawa) [reply contrib/test/ci/build/kata.yml]: Fixed. Thanks ! :-)

**Comment by littlejawa** [comment]
Update:
- move to kata 3.6.0
- enable hooks tests, as the fix for it was released with 3.5.0
- add some kata test specific enhancements (enable debug, cleanup lingering processes)
- add an integration test specific to kata, testing we're actually using the kata runtime. Useful to validate config/script changes in the CI

**Comment by littlejawa** [comment]
@kwilczynski - sorry for pinging you again on this one, but I've updated it and it needs approval again.

Note that I have modified some settings for the kata test, but they have no impact on other CI jobs.

**Comment by kwilczynski** [comment]
@cri-o/cri-o-maintainers, please help a look and help us with some final approvals, thank you!

**Comment by littlejawa** [comment]
@cri-o/cri-o-maintainers - I think this one is ready to merge. The remaining failure on fedora-integration is unrelated as far as I can tell.

File: test/kata.bats
Comment (saschagrunert) [inline test/kata.bats:24]: Can we remove that?

**Review by saschagrunert — APPROVED** [review]

**Comment by littlejawa** [comment]
@saschagrunert - remaining error in cgroupv2-integration is unrelated (see #8389)
And I know you're working on fixing the RPM builds :-)
Do you think we can merge it?

**Comment by saschagrunert** [comment]
Merging, the kata tests were successful.

**Pull request merged**
